lib/pacemaker/libpacemaker_private.h

```diff
@@ -25,10 +25,9 @@ struct resource_alloc_functions_s {
     gboolean(*create_probe) (pe_resource_t *, pe_node_t *, pe_action_t *, gboolean, pe_working_set_t *);
     void (*internal_constraints) (pe_resource_t *, pe_working_set_t *);
 
-    void (*rsc_colocation_lh) (pe_resource_t *, pe_resource_t *,
-                               pcmk__colocation_t *);
-    void (*rsc_colocation_rh) (pe_resource_t *, pe_resource_t *,
-                               pcmk__colocation_t *);
+    void (*apply_coloc_score) (pe_resource_t *dependent, pe_resource_t *primary,
+                               pcmk__colocation_t *constraint,
+                               bool for_dependent);
```

Comment:
A bound method would be neat here but this is C.

In that line of thinking, I tossed around the idea that this might read more intuitively, since our boolean really answers the question of "which resource" rather than "yes/no":
```
    void (*apply_coloc_score) (pe_resource_t *rsc, pe_resource_t *dependent,
                               pe_resource_t *primary,
                               pcmk__colocation_t *constraint);
```
and then have the function compare pointers to determine dependent vs. primary. (Check for edge case where `dependent == primary`.)

Of course it doesn't actually matter. The boolean works great and does the job of merging.